app/src/main/java/at/aau/serg/logic/StoreToken.kt

```diff
@@ -2,18 +2,45 @@ package at.aau.serg.logic
 
 import android.content.ContextWrapper
 import org.json.JSONException
+import org.json.JSONObject
 
-class StoreToken {
-    fun storeTokens(accessToken: String, refreshToken: String, context: ContextWrapper, secret: Secret){
-        try{
-            val sharedPreferences =secret.getSecretSharedPref(context)
-            val editor = sharedPreferences.edit()
-            editor.putString("accessToken", accessToken)
-            editor.putString("refreshToken", refreshToken)
-            editor.apply()
-        }catch (e: JSONException) {
-            e.printStackTrace()
-        }
+class StoreToken(private val context: ContextWrapper, private val secret: Secret) {
```

Comment:
Am I correct that we currently have only one secret? If so, is it necessary that we provide it as a parameter in the constructor every time we use `StoreToken`? It would be cleaner to just omit it and fetch it in an init block.

Reply:
I have changed it now.
For the test cases I have created a setter for the Secret to pass the Mockked Secret

Reply:
We should be able to mock the `Secret` function. Please take a look at my comment below.